lib/iris/analysis/stats.py

```diff
@@ -94,7 +95,29 @@ def _get_calc_view(cube_a, cube_b, corr_coords):
     reshaped_b = data_b.transpose(slice_ind+res_ind)\
                        .reshape(dim_i_len, dim_j_len)
 
-    return reshaped_a, reshaped_b, res_ind
+    # Remove data where one or both cubes are masked
+    # First deal with the case that either cube is unmasked
+    # Collapse masks to the dimension we are correlating over (0th)
+    if np.ma.is_masked(reshaped_a):
+        a_not_masked = np.logical_not(reshaped_a.mask).any(axis=1)
+    else:
+        a_not_masked = True
+    if np.ma.is_masked(reshaped_b):
+        b_not_masked = np.logical_not(reshaped_b.mask.any(axis=1))
+    else:
+        b_not_masked = True
+
+    both_not_masked = a_not_masked & b_not_masked
+    try:
+        # compress to good values using mask array
+        return_a = reshaped_a.compress(both_not_masked)
+        return_b = reshaped_b.compress(both_not_masked)
+    except ValueError:
+        # expect when masks are just non-array True/False
```

Comment:
please could you expand the comment a little to explain why any ValueError is acceptable to be excepted, I feel I am missing a bit of logic